Sources/NIOCore/ChannelPipeline.swift

```diff
@@ -1574,6 +1626,38 @@ extension ChannelPipeline {
     }
 }
 
+extension ChannelPipeline.SynchronousOperations {
+    /// A `Position` within the `ChannelPipeline`'s `SynchronousOperations` used to insert non-sendable handlers
+    /// into the `ChannelPipeline` at a certain position.
+    @preconcurrency
+    public enum Position {
+        /// The first `ChannelHandler` -- the front of the `ChannelPipeline`.
+        case first
+
+        /// The last `ChannelHandler` -- the back of the `ChannelPipeline`.
+        case last
+
+        /// Before the given `ChannelHandler`.
+        case before(ChannelHandler)
+
+        /// After the given `ChannelHandler`.
+        case after(ChannelHandler)
+
+        package init(_ position: ChannelPipeline.Position) {
```

Comment:
I can't really see any reason to not make this `public`: we may need this conversion logic in other places.